tests/integration/run-prettier.js

```diff
@@ -40,12 +40,21 @@ async function run(dir, args, options) {
 
   jest.spyOn(Date, "now").mockImplementation(() => 0);
 
+  const orig_writeFile = fs.promises.writeFile;
   const write = [];
 
   jest
     .spyOn(fs.promises, "writeFile")
-    // eslint-disable-next-line require-await
-    .mockImplementation(async (filename, content) => {
+    .mockImplementation(async function(filename, content) {
```

Comment:
This is good as it is, the problem is in `next` branch, we don't relay on `mockImplementation`, do you think it's possible to use a simple option like `{mockWriteFileError: 'error message'}`, so we can pass it to the worker? Otherwise this will be hard when we merge into the `next` branch

https://github.com/prettier/prettier/blob/166138cfddcc3f5ecab9b366e35270d0c3f79d0f/tests/integration/cli-worker.js#L12-L17

Reply:
Or maybe `{mockWriteFileError: {'a.js': 'error message'}}` in this case.